spec/GraphQLOverHTTP.md

```diff
@@ -112,72 +178,152 @@ http://example.com/product/graphql
 
 # Serialization Format
 
-The GraphQL specification allows for many [serialization formats to be implemented](https://spec.graphql.org/October2021/#sec-Serialization-Format). Servers and clients over HTTP MUST support JSON and MAY support other, additional serialization formats.
+The GraphQL specification allows for many
+[serialization formats to be implemented](https://spec.graphql.org/draft/#sec-Serialization-Format).
+Servers and clients MUST support JSON and MAY support other, additional
+serialization formats.
+
+For consistency and ease of notation, examples of the response are given in JSON
+throughout this specification.
+
+## Media Types
+
+The following are the officially recognized GraphQL media types to designate
+using the JSON encoding for GraphQL requests and responses over HTTP.
+
+| Name                       | Description                                   |
+| -------------------------- | --------------------------------------------- |
+| `application/graphql+json` | The preferred type; better HTTP compatibility |
+| `application/json`         | To support legacy clients                     |
+
+For details of the shapes of these JSON payloads, please see
+[Request](#sec-Request) and [Response](#sec-Response).
+
+If the media type in a `Content-Type` or `Accept` header includes encoding
+information, then the encoding MUST be `utf-8` (e.g.
+`Content-Type: application/graphql+json; charset=utf-8`). If encoding
+information is not included then `utf-8` should be assumed.
+
+A server MUST support requests encoded with the `application/json` media type
+(as indicated by the `Content-Type` header).
+
+A server MUST support requests which accept the `application/json` media type
+(as indicated by the `Accept` header).
+
+A server SHOULD support requests encoded with the `application/graphql+json`
+media type (as indicated by the `Content-Type` header).
 
-For consistency and ease of notation, examples of the response are given in JSON throughout the spec.
+A server SHOULD support requests which accept the `application/graphql+json`
+media type (as indicated by the `Accept` header).
 
-## Content Types
+A server MAY support requests encoded with and/or accepting other media types.
 
-The following are the officially recognized GraphQL content types to designate encoding JSON over HTTP.
+If the client does not supply a `Content-Type` header, the server SHOULD treat
```

Comment:
Do we have servers out there that accept the `Content-Type` being empty? I think the biggest issue was more about the Accept header since many clients do not yet send one. 

I think we can expect the `Content-Type` header to be set to `application/json` or `application/graphql+json`. Legacy clients will send `application/json` and this SHOULD be supported.

Reply:
You're probably right; even if you're using `curl` you'd expect to set a content-type header, otherwise it's assumed it's `application/x-www-form-urlencoded`. Good catch; I'll get this fixed 👍

Reply:
I've changed it such that the `Content-Type` **MUST** be indicated by the client.